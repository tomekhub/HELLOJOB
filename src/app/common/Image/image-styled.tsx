import styled from 'styled-components';

const Element = styled.img``;

export { Element };
