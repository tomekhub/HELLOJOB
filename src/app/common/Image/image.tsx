import React from 'react';
import { Element } from './image-styled';

interface ImageProps {
  alt?: string;
  src: string;
  className?: string;
  classNameActive?: string;
}

const CommonImage: React.FunctionComponent<ImageProps> = ({
  alt = 'Image',
  src,
  ...rest
}): JSX.Element => {
  return <Element alt={alt} src={src} {...rest} />;
};

export default CommonImage;
