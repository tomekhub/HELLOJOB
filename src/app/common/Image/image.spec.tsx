import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Image from './image';
import renderer from 'react-test-renderer';

describe('common Image', () => {
  let ImageComponent: ShallowWrapper;

  beforeEach(() => {
    ImageComponent = shallow(<Image src="imageUrl" />);
  });

  it('Image render with with src equal prop: imageUrl', () => {
    expect(ImageComponent.prop('src')).toEqual('imageUrl');
  });

  it('Image render with with default alt : Image', () => {
    expect(ImageComponent.prop('alt')).toEqual('Image');
  });

  it('Image render with with given alt prop: Logo', () => {
    ImageComponent.setProps({ alt: 'Logo' });
    expect(ImageComponent.prop('alt')).toEqual('Logo');
  });

  it('Image renders correctly', () => {
    const tree = renderer.create(<Image src="xd" alt="alert" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
