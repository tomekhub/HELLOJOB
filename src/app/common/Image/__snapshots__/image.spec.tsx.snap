// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common Image Image renders correctly 1`] = `
<img
  alt="alert"
  className="sc-bdVaJa bDWFJH"
  src="xd"
/>
`;
